pkg/api/types.go

```diff
@@ -747,6 +747,9 @@ type Endpoints struct {
 	TypeMeta   `json:",inline"`
 	ObjectMeta `json:"metadata,omitempty"`
 
+	// Optional: The IP protocol for these endpoints. Supports "TCP" and
+	// "UDP".  Defaults to "TCP".
+	Protocol  Protocol `json:"protocol,omitempty"`
```

Comment:
#4370 is the destination.  I'm trying to break the implementation into discrete steps.  I'm OK with just doing commits in one massive PR, if people prefer.  I just hate reviewing those...